### PR TITLE
added documentation for Assignment class in sympy/codegen/ast.py

### PR DESCRIPTION
**Pull request by avishrivastava11**
Documentation for `Assignment` class regarding indexed docs has been added using examples, highlighting the inclusion of end points while indexing for languages like `C` and `Fortran`.

#12781
Fix for issue #12781

Contributers, members, others please review this PR and let me know the improvements.
Any feedback is highly appreciated.
Thanks.

### CODE REVIEW

**Comment by avishrivastava11** [comment]
@asmeurer 

adding the `ccode` and `fcode` is giving this error in Travis CI :-

```
Warning, treated as error:
/home/travis/build/sympy/sympy/sympy/codegen/ast.py:docstring of sympy.codegen.ast.Assignment:21:Error in "rubric" directive:
no content permitted.
.. rubric:: Examples
    >>> from sympy import symbols, MatrixSymbol, Matrix
    >>> from sympy.codegen.ast import Assignment
    >>> x, y, z = symbols('x, y, z')
    >>> Assignment(x, y)
    Assignment(x, y)
    >>> Assignment(x, 0)
    Assignment(x, 0)
    >>> A = MatrixSymbol('A', 1, 3)
    >>> mat = Matrix([x, y, z]).T
    >>> Assignment(A, mat)
    Assignment(A, Matrix([[x, y, z]]))
    >>> Assignment(A[0, 1], x)
    Assignment(A[0, 1], x)
    >>> n = symbols('n', integer=True)
    >>> ccode(Assignment(Indexed("A", Idx('i', (0, n))), 0))  #case_1
    'for (int i=0; i<n + 1; i++){
   A[i] = 0;
make: *** [html] Error 2
```

any suggestions which can be done to avoid this?

**Comment by avishrivastava11** [comment]
also @asmeurer
although the terminal shows the whitespaces on running
`fcode(Assignment(Indexed("A", Idx('i', (0, n))), 0))`
output == >  `'      do i = 1, n + 1\n         A(i) = 0\n      end do'`
Travis CI is resulting in an error showing `leading whitespaces`.
what should I do to make it pass those tests?

File: sympy/codegen/ast.py [hunk 1]
```diff
@@ -106,6 +106,28 @@ class Assignment(Relational):
...
+    >>> from sympy import ccode
+    >>> from sympy import fcode
+    >>> from sympy.tensor import Indexed
+    >>> from sympy.tensor import Idx
+
+
+    >>> n = symbols('n', integer=True)
+
+    >>> ccode(Assignment(Indexed("A", Idx('i', (0, n))), 0)) #case_1
+    'for (int i=0; i<n + 1; i++){\n   A[i] = 0;\n}'
```
[the remark this answers is not in the document]
Comment (asmeurer) [reply sympy/codegen/ast.py]: Or you can add an `r` to the beginning of the docstring, like `r"""`

[the remark this answers is not in the document]
Comment (avishrivastava11) [reply sympy/codegen/ast.py]: @jksuom 
@asmeurer as you suggested 
>Or you can add an `r` to the beginning of the docstring, like `r"""`

I have added `r` to the beginning of the docstring.
pls review.

**Comment by avishrivastava11** [comment]
@asmeurer @jksuom  and other members, pls review and If there are no further changes, please merge this PR.

File: sympy/codegen/ast.py [hunk 2]
```diff
@@ -106,6 +106,28 @@ class Assignment(Relational):
...
+
+    >>> fcode(Assignment(Indexed("A", Idx('i', (0, n))), 0)) #case_2
+    '      do i = 1, n + 1\n         A(i) = 0\n      end do'
+
+    As shown in case_1, the C loop runs from [0, n], i.e. the upper limit is
```
Comment (sidhantnagpal) [inline sympy/codegen/ast.py]: The explanation can be accompanied with the corresponding doctest for brevity.

Comment (avishrivastava11) [reply sympy/codegen/ast.py]: @sidhantnagpal  pls take a look now. I've added the changes you suggested.

**Comment by avishrivastava11** [comment]
@asmeurer  Can you review this PR?

**Comment by avishrivastava11** [comment]
@Abdullahjavednesar  I think the `author's turn` is misleading since I have incorporated all the changes everyone asked for. Please remove that label. Thanks

**Comment by avishrivastava11** [comment]
@certik Review please

**Comment by avishrivastava11** [comment]
@jksuom @smichr  Review please.

**Pull request closed without merging**